### PR TITLE
Drop server.R template file

### PR DESCRIPTION
**Pull request by schloerke**
Cleaner history of https://github.com/rstudio/shiny/pull/2825. Closes #2825

Should be merged into #2826 before #2826 is merged into master

* Produces a full `adhoc.R` file that uses `withr::with_environment()` + `loadSupport()` and has dynamic content.
* Removed `helper-support.R` as `testthat::test_dir` now calls `loadSupport` directly
* Removed regular shiny package testing output
* Fixed broken lexical sort test
* Added context to all `testthat` files

### CODE REVIEW

**Comment by schloerke** [comment]
Appveyor is failing due to glue being updated. Travis is passing

File: DESCRIPTION [hunk 1]
```diff
@@ -92,12 +92,13 @@ Suggests:
     ggplot2,
     reactlog (>= 1.0.0),
     magrittr,
-    shinytest,
+    shinytest (> 1.3.1),
```
Comment (wch) [inline DESCRIPTION:95]: This doesn't do anything in Suggests. We should add a version check in `shinyAppTemplate()`.

File: R/app_template.R [hunk 2]
```diff
@@ -153,13 +151,27 @@ shinyAppTemplate <- function(path = NULL, examples = "default")
...
     }
 
-    file.copy(
-      file.path(example_path("tests"), files),
-      file.path(path, "tests", files)
-    )
+    for (file in files) {
+      from_file <- file.path(example_path("tests"), file)
+      to_file <- file.path(path, "tests", file)
+      # is glue file
```
Comment (wch) [inline R/app_template.R:160]: Technically, not glue, but htmlTemplate

File: tests/testthat/test-test.R [hunk 3]
```diff
@@ -143,3 +134,53 @@ test_that("runTests runs as expected without rewiring", {
   ))
   expect_s3_class(df, "shiny_runtests")
 })
+
+
+context("shinyAppTemplate + runTests")
+test_that("app template works with runTests", {
+
+  testthat::skip_if_not_installed("shinytest", "1.3.1.9000")
+
```
Comment (wch) [inline tests/testthat/test-test.R:143]: Skip on CRAN.

File: tests/testthat/test-test.R [hunk 4]
```diff
@@ -143,3 +134,53 @@ test_that("runTests runs as expected without rewiring", {
...
+    tempTemplateDir <- file.path(tempdir(), random_folder)
+    shinyAppTemplate(tempTemplateDir, combo)
+    on.exit(unlink(tempTemplateDir, recursive = TRUE))
+
+    if (any(c("all", "shinytest", "testthat") %in% combo)) {
+      expect_output(
+        print(runTests(tempTemplateDir)),
+        paste0(
+          "Shiny App Test Results\\n\\* Success",
+          if ("shinytest" %in% combo) paste0("\\n  - ", file.path(random_folder, "tests", "shinytest\\.R")),
```
Comment (wch) [inline tests/testthat/test-test.R:173]: Should include `"all"`

**Comment by schloerke** [comment]
Appveyor is failing due to glue being updated. Travis is passing

Merging.

**Pull request merged**